lib/cli/ui/ansi.rb

```diff
@@ -42,7 +42,7 @@ def printing_width(str)
         #
         #: (String str) -> String
         def strip_codes(str)
-          str.gsub(/\x1b\[[\d;]+[A-Za-z]|\x1b\][\d;]+.*?\x1b\\|\r/, '')
+          str.gsub(/\x1b\[[\d;]+[A-Za-z]|\x1b\][\d;]+[^\x1b\x07]*(\x1b\\|\x07)|\r/, '')
```

Comment:
Maybe we could do something like:

```ruby
        # https://ghostty.org/docs/vt/concepts/sequences#csi-sequences
        CSI_SEQUENCE = /\x1b\[[\d;:]+[\x20-\x2f]*?[\x40-\x7e]/
        # https://ghostty.org/docs/vt/concepts/sequences#osc-sequences
        OSC_SEQUENCE = /\x1b\][^\x00-\x1f]*?\x1b\x5c/

...

          str.gsub(Regexp.union(CSI_SEQUENCE, OSC_SEQUENCE, /\r/), '')
```

I found this hard to decipher and I think this would help. LGTM otherwise